src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/CacheEntry.java

```diff
@@ -0,0 +1,382 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.http_cache.internal.stream.util;
+
+import static java.lang.Integer.MAX_VALUE;
+import static java.lang.Integer.parseInt;
+import static org.reaktivity.nukleus.http_cache.internal.stream.util.CacheDirectives.MAX_AGE;
+import static org.reaktivity.nukleus.http_cache.internal.stream.util.CacheDirectives.MAX_STALE;
+import static org.reaktivity.nukleus.http_cache.internal.stream.util.CacheDirectives.MIN_FRESH;
+import static org.reaktivity.nukleus.http_cache.internal.stream.util.CacheDirectives.S_MAXAGE;
+import static org.reaktivity.nukleus.http_cache.internal.stream.util.HttpCacheUtils.sameAuthorizationScope;
+import static org.reaktivity.nukleus.http_cache.internal.stream.util.HttpHeaders.CACHE_CONTROL;
+import static org.reaktivity.nukleus.http_cache.internal.stream.util.HttpHeaders.WARNING;
+import static org.reaktivity.nukleus.http_cache.internal.stream.util.HttpHeadersUtil.getHeader;
+
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.Date;
+import java.util.function.Consumer;
+
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.reaktivity.nukleus.function.MessageConsumer;
+import org.reaktivity.nukleus.http_cache.internal.Correlation;
+import org.reaktivity.nukleus.http_cache.internal.types.HttpHeaderFW;
+import org.reaktivity.nukleus.http_cache.internal.types.ListFW;
+import org.reaktivity.nukleus.http_cache.internal.types.OctetsFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.EndFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.ResetFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.WindowFW;
+
+public final class CacheEntry
+{
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss zzz");
+
+    private final Cache cache;
+    private final int requestSlot;
+    private final int requestSize;
+    private final int responseSlot;
+    private final int responseHeaderSize;
+    private final int responseSize;
+    private int clientCount = 0;
+    private boolean cleanUp = false;
+
+    private Instant lazyInitiedResponseReceivedAt;
+    private Instant lazyInitiedResponseStaleAt;
+
+    CacheEntry(
+        Cache cache, int requestSlot,
+        int requestSize,
+        int responseSlot,
+        int responseHeaderSize,
+        int responseSize)
+    {
+        this.cache = cache;
+        this.requestSlot = requestSlot;
+        this.requestSize = requestSize;
+        this.responseSlot = responseSlot;
+        this.responseHeaderSize = responseHeaderSize;
+        this.responseSize = responseSize;
+    }
+
+    private void handleEndOfStream(
+            int msgTypeId,
+            DirectBuffer buffer,
+            int index,
+            int length)
+    {
```

Comment:
I wonder what is the usage of these signatures if they are not being used.

Reply:
removed.